File: test_upload.py
import requests
import base64
import json

# 创建一个最小的PNG文件
def create_test_png():
    # 最小的PNG文件（1x1像素，黑色）
    png_data = base64.b64decode(
        b'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg=='
    )
    with open('test.png', 'wb') as f:
        f.write(png_data)
    print("Created test.png file")

# 测试上传功能
def test_upload():
    create_test_png()
    
    url = 'http://localhost:8080/api/outfit/upload'
    files = {'files': open('test.png', 'rb')}
    
    try:
        print("Sending request to", url)
        response = requests.post(url, files=files)
        print(f"Response status code: {response.status_code}")
        print(f"Response headers: {dict(response.headers)}")
        
        try:
            json_response = response.json()
            print(f"Response JSON: {json.dumps(json_response, indent=2, ensure_ascii=False)}")
        except json.JSONDecodeError:
            print(f"Response content: {response.text}")
            
    except Exception as e:
        print(f"Error occurred: {e}")

if __name__ == "__main__":
    test_upload()

File: TestUpload.java
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.mime.MultipartEntityBuilder;
import org.apache.http.impl.client.HttpClients;
import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.util.Scanner;

public class TestUpload {
    public static void main(String[] args) throws IOException {
        // 创建一个空的PNG文件用于测试
        File testFile = new File("test.png");
        if (!testFile.exists()) {
            // 创建一个最小的PNG文件
            String pngHeader = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==";
            byte[] pngBytes = java.util.Base64.getDecoder().decode(pngHeader);
            java.nio.file.Files.write(testFile.toPath(), pngBytes);
            System.out.println("创建了测试图片文件: " + testFile.getAbsolutePath());
        }

        HttpClient httpclient = HttpClients.createDefault();
        HttpPost httppost = new HttpPost("http://localhost:8080/api/outfit/upload");

        MultipartEntityBuilder builder = MultipartEntityBuilder.create();
        builder.addBinaryBody("files", testFile);
        httppost.setEntity(builder.build());

        System.out.println("正在发送请求...");
        HttpResponse response = httpclient.execute(httppost);
        System.out.println("响应状态: " + response.getStatusLine());

        // 读取响应内容
        try (InputStream inputStream = response.getEntity().getContent()) {
            Scanner scanner = new Scanner(inputStream, "UTF-8");
            String responseBody = scanner.useDelimiter("\\A").next();
            System.out.println("响应内容: " + responseBody);
        }

        // 清理
        testFile.delete();
    }
}